tasks/pre_checks/validate_data_center_name.yml

```diff
@@ -0,0 +1,14 @@
+- name: Validate Data Center name format
+  block:
+  - name: Fail if Data Center name format is incorrect
+    fail:
+      msg: >-
+        "Invalid Data Center name format. Data Center name may only contain letters, numbers, '-', or '_'."
+        " Got {{ he_data_center }}"
+    when: not he_data_center | regex_search( "^[a-zA-Z0-9_-]+$" )
+  - name: Validate Cluster name
```

Comment:
I'd suggest to also check this for invalid chars as for the datacenter name and move here the check we already have in:
https://github.com/oVirt/ovirt-ansible-hosted-engine-setup/blob/master/tasks/bootstrap_local_vm/05_add_host.yml#L65

Reply:
why do you want to move the validation to the middle of the deployment? the whole point is to fail on the beginning

Reply:
I'm suggesting to anticipate here the check that now is in https://github.com/oVirt/ovirt-ansible-hosted-engine-setup/blob/master/tasks/bootstrap_local_vm/05_add_host.yml#L65https://github.com/oVirt/ovirt-ansible-hosted-engine-setup/blob/master/tasks/bootstrap_local_vm/05_add_host.yml#L65

Reply:
OK, sorry. In this stage we don't have any running engine so we cannot do more than this